inductiva/tasks/run_simulation.py

```diff
@@ -39,4 +45,26 @@ def run_simulation(
         raise RuntimeError(
             f"Expected result to be a string with task_id, got {type(task_id)}")
 
+    if not format_utils.getenv_bool("DISABLE_TASK_METADATA_LOGGING", False):
+        metadata = {
+            **{
+                "task_id": task_id,
+                "input_dir": str(input_dir),
+                "storage_dir": str(storage_dir),
+                "api_method_name": api_method_name,
+                "machine_group_id": resource_pool_id,
+            },
+            **kwargs,
+        }
+        _save_metadata(metadata)
+
     return task
+
+
+def _save_metadata(metadata):
+    """Appends metadata to the TASK_METADATA_FILENAME in the cwd."""
+    file_path = pathlib.Path().cwd() / TASK_METADATA_FILENAME
+    with open(file_path, "a", encoding="utf-8") as f:
+        json.dump(metadata, f)
+        f.write("\n")
+    logging.info("Simulation metadata logged to: %s", file_path)
```

Comment:
This is unnecessary now, since all metadata is always saved to the same file in the current working directory. As long as it will be explained in the documentation it is enough for users to grasp things.

Reply:
I would prefer to leave this here because the behaviour can be deactivated. Therefore, it would be nice to have some type of logs of stating whether or not metadata is being saved